### PR TITLE
 [plugins/aws][feat] Add support for Elastic Beanstalk Application Environments and Resources

### PR DESCRIPTION
**Pull request by anjafr**
# Description

- collect existing tags on Elasticbeanstalk Applications
- collect Environments of Applications (incl. existing tags)
- add edges from Environments to Applications
- collect Resources in Environments
- add edges from Resources to Environments
- tagging and deletion methods for Environments

# To-Dos

- [x] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`
- [ ] Document new or updated functionality (someengineering/resoto.com#XXXX)

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: plugins/aws/resoto_plugin_aws/resource/elasticbeanstalk.py [hunk 1]
```diff
@@ -103,4 +122,214 @@ def delete_resource(self, client: AwsClient) -> bool:
...
+        res = self.resources
+        if not res:
+            return
+        if res.auto_scaling_groups:
+            for group in res.auto_scaling_groups:
+                if group.auto_scaling_group_name:
+                    builder.dependant_node(
+                        self,
+                        reverse=True,
+                        # delete_same_as_default=True, ??
```
Comment (meln1k) [inline plugins/aws/resoto_plugin_aws/resource/elasticbeanstalk.py:281]: do we need this line?

**Review by meln1k — COMMENTED** [review]
small nit, lgtm otherwise

File: plugins/aws/resoto_plugin_aws/resource/elasticbeanstalk.py (hunk 1 above)
Comment (anjafr) [reply plugins/aws/resoto_plugin_aws/resource/elasticbeanstalk.py:281]: I don't know - hence the questionmarks 💡 `delete_same_as_default` is `False` per default. But I don't know if it should be `True` here?

Comment (meln1k) [reply plugins/aws/resoto_plugin_aws/resource/elasticbeanstalk.py:281]: I think if it works on the live cloud, then we don't need it, right?

**Review by meln1k — DISMISSED** [review]
LGTM! thanks!

**Review by meln1k — APPROVED** [review]
LGTM

**Pull request merged**